base/ltcmd.dtx

```diff
@@ -435,18 +435,75 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}[EXP]{\@@_all_m_check:n, \@@_all_m_check_aux:n}
+%   A quick loop to check for all |(+)m|-type arguments.
+%    \begin{macrocode}
+\cs_new:Npn \@@_all_m_check:n #1
+  { \tl_map_function:nN {#1} \@@_all_m_check_aux:n }
+\cs_new:Npn \@@_all_m_check_aux:n #1
+  {
+    \str_if_eq:nnF {#1} { m }
+      {
+        \str_if_eq:nnF {#1} { + }
+          { X }
+      }
+  }
+%    \end{macrocode}
+% \end{macro}
+%
 % \begin{macro}{\@@_declare_cmd_code:Nnn}
 % \begin{macro}
-%   {\@@_declare_cmd_code_aux:Nnn, \@@_declare_cmd_code_expandable:Nnn}
-%   The appropriate auxiliary is called.
+%   {
+%     \@@_declare_cmd_optimised:Nnn,
+%     \@@_declare_cmd_code_aux:Nnn,
+%     \@@_declare_cmd_code_expandable:Nnn
+%   }
+% \begin{macro}{\@@_start_optimised:}
+% \changes{v1.2b}{2023/11/28}
+%   {Optimise cmd creation for all-\texttt{m} arguments}
+%   At this stage we can check for a short-cut possibility: if the signature
+%   is just made up of \cs{@@_expandable_grab_m:w}, then we can produce an
+%   optimised document command. This only applies to document commands,
+%   not creation of environments (which are more complex).
 %    \begin{macrocode}
-\cs_new_protected:Npn \@@_declare_cmd_code:Nnn
+\cs_new_protected:Npn \@@_declare_cmd_code:Nnn #1#2
   {
-    \bool_if:NTF \l_@@_grab_expandably_bool
-      { \@@_declare_cmd_code_expandable:Nnn }
-      { \@@_declare_cmd_code_aux:Nnn }
+    \bool_lazy_all:nTF
+      {
+        { ! \l_@@_environment_bool }
+        {
+          ! \bool_lazy_and_p:nn
+            { \l_@@_some_short_bool }
+            { \l_@@_some_long_bool }
+        }
+        { \tl_if_blank_p:e { \@@_all_m_check:n {#2} } }
+      }
+      { \@@_declare_cmd_optimised:Nnn }
+      {
+        \bool_if:NTF \l_@@_grab_expandably_bool
+          { \@@_declare_cmd_code_expandable:Nnn }
+          { \@@_declare_cmd_code_aux:Nnn }
+      }
```

Comment:
Only one branch of bool-expr is not flipped, would `\bool_lazy_or:nTF { {expr 1} {expr 2} {! expr_3} } {<normal>} {<optimized>}` be a little quicker (and more straight)?
```suggestion
    \bool_lazy_or:nTF
      {
        { \l_@@_environment_bool }
        {
          \bool_lazy_and_p:nn
            { \l_@@_some_short_bool }
            { \l_@@_some_long_bool }
        }
        { ! \tl_if_blank_p:e { \@@_all_m_check:n {#2} } }
      }
      {
        \bool_if:NTF \l_@@_grab_expandably_bool
          { \@@_declare_cmd_code_expandable:Nnn }
          { \@@_declare_cmd_code_aux:Nnn }
      }
      { \@@_declare_cmd_optimised:Nnn }
```